### ISSUE TITLE
[Api] Submit and verify CSRF token with all API requests

### ISSUE THREAD

**Issue by lukasbestle** [getkirby/kirby#800]
See #791.

**Comment by lukasbestle** [comment]
Hm, actually that's quite complex. A REST API is supposed to be stateless, which doesn't work with CSRF tokens.

Of course security is more important here. The only reason we have this issue at all is that we use sessions for authentication. But for client-side applications that's the only secure way of handling auth to avoid storing the credentials in client-side storage, so I still think it's the best solution for the Panel.

Basically we need to add CSRF token verification to all API requests that use a session for auth. We also currently support Basic auth which should not require CSRF tokens as that one is intended to be used outside of the browser. Am I right @mzur?

**Comment by mzur** [comment]
Right. Stateless REST is only possible with an authentication token which should be equivalent to basic auth over HTTPS. All authentication using cookies/sessions needs CSRF protection.

**Comment by bastianallgeier** [comment]
✅
